File: tools_box/tools_box/report/purchase_by_items_report_summary/purchase_by_items_report_summary.js
// Copyright (c) 2016, bobzz.zone@gmail.com and contributors
// For license information, please see license.txt

frappe.query_reports["Purchase by items Report Summary"] = {
	"filters": [
		{
			"fieldname":"from",
			"label": __("From Date"),
			"fieldtype": "Date",
			"width": "80",
			"reqd":1
		},
		{
			"fieldname":"to",
			"label": __("To Date"),
			"fieldtype": "Date",
			"default": get_today(),
			"reqd":1
		},
		{
			"fieldname":"supplier",
			"label": __("Supplier"),
			"fieldtype": "Link",
			"options": "Supplier",
			"reqd":0
		},
		{
			"fieldname":"item",
			"label": __("Item"),
			"fieldtype": "Link",
			"options": "Item",
			"reqd":0
		}
	]
}

File: tools_box/tools_box/report/purchase_by_items_report_summary/purchase_by_items_report_summary.py
# Copyright (c) 2013, bobzz.zone@gmail.com and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = ["Item:Link/Item:200","Item Name:Data:200","Qty:Float:200","Amount:Currency:200"], []
	item=""
	supplier=""
	territory=""
	if filters.get("item"):
		item = """ and soi.item_code = '{}' """.format(filters.get("item"))
	if filters.get("supplier"):
		supplier = """ and so.supplier = '{}' """.format(filters.get("supplier"))
	data = frappe.db.sql("""select soi.item_code,soi.item_name,sum(soi.qty),sum(soi.amount) from `tabPurchase Invoice Item` soi 
		join `tabPurchase Invoice` so on soi.parent=so.name where so.docstatus=1 and (so.posting_date between '{}' and '{}') {} {} group by soi.item_code""".format(filters.get("from"),filters.get("to"),item,supplier),as_list=1 )
	return columns, data
